### DISCUSSION TITLE
Magneto Control

### DISCUSSION THREAD

**Discussion by RobSimonton** [nguyenquyhy/Flight-Tracker-StreamDeck#157] [Q&A]
Just new to Flight-Tracker-Streamdeck but having good luck and fun.
I am currently working with the magnetos and having limited success.
The magneto_start event works on a button press.
The magneto_off event kind of works, the first press the key goes from Both to Left then back to Both, the second press the key goes to off and the engine stops.
The _LEFT, _RIGHT, _BOTH, _INCR, and DECR do not seem to do anything.
I am using the un-number Magneto events, the Magneto1 events for the above did not work either.
I am using the 172S G1000 as a testbed.

Thanks

**Comment by rmroc451** [comment]
> Just new to Flight-Tracker-Streamdeck but having good luck and fun.
> I am currently working with the magnetos and having limited success.
> The magneto_start event works on a button press.
> The magneto_off event kind of works, the first press the key goes from Both to Left then back to Both, the second press the key goes to off and the engine stops.
> The _LEFT, _RIGHT, _BOTH, _INCR, and DECR do not seem to do anything.
> I am using the un-number Magneto events, the Magneto1 events for the above did not work either.
> I am using the 172S G1000 as a testbed.
> Thanks

I've seen that before too, in my mag switch implementation. 

If you fire them in the order the switch would travel, it works better. You'd have to hit the buttons in the right order based on which state it is in.

I use those same events for my arduino connected mag key switch. Since my switch travels the range and fires each event as the switch moves, I get the correct movement in the sim.

Without making a mag button to simulate that, you are kinda SOL :(

**Reply by RobSimonton** [reply]
ok, thanks
